src/listeners/grafana/pagerduty/handle-grafana-pagerduty-alerts.test.js

```diff
@@ -0,0 +1,187 @@
+import {
+  createDedupeKey,
+  findIntegrationKeyForService,
+  getTeams,
+  handleGrafanaPagerDutyAlert,
+  shouldSendAlert
+} from '~/src/listeners/grafana/pagerduty/handle-grafana-pagerduty-alerts.js'
+import { serviceToPagerDutyServiceOverride } from '~/src/config/pagerduty-service-override.js'
+import { pino } from 'pino'
+import { fetchService } from '~/src/helpers/fetch/fetch-service.js'
+import { logger } from '@azure/identity'
+import { config } from '~/src/config/index.js'
+import { sendAlert } from '~/src/helpers/pagerduty/send-alert.js'
+
+describe('#createDedupeKey', () => {
+  test('should create the same md5 string for identical payloads', () => {
+    const payload = {
+      summary: 'test',
+      service: 'foo',
+      environment: 'dev',
+      startsAt: '2025-02-03T11:14:25.507Z'
+    }
+    expect(createDedupeKey(payload)).toEqual(createDedupeKey(payload))
+  })
+
+  test('should handle malformed payloads gracefully', () => {
+    const malformed = {
+      summary: 'test',
+      service: 'foo',
+      startsAt: '2025-02-03T11:14:25.507Z'
+    }
+
+    expect(createDedupeKey(malformed)).not.toBeNull()
+    expect(createDedupeKey(null)).not.toBeNull()
+  })
+})
+
+describe('#shouldSendAlert', () => {
+  test('should only sends alerts for configured environments', () => {
+    const alert = {
+      environment: '',
+      team: 'test',
+      service: 'test-service',
+      alertName: 'test-alert',
+      status: 'critical',
+      startsAt: '2025-02-03T11:14:25.507Z',
+      endsAt: '2025-02-03T11:14:25.507Z',
+      summary: 'an alert happened',
+      description: 'it broke',
+      series: '1',
+      runbookUrl: '',
+      alertURL: ''
+    }
+    const envsToAlert = ['prod', 'management']
+    const envsToNotAlert = ['dev', 'test', 'infra-dev']
+
+    for (const env of envsToAlert) {
+      alert.environment = env
+      expect(shouldSendAlert(alert, envsToAlert)).toBe(true)
+    }
+
+    for (const env of envsToNotAlert) {
+      alert.environment = env
+      expect(shouldSendAlert(alert, envsToAlert)).toBe(false)
+    }
+  })
+
+  test('should override environments from config', () => {
+    const alert = {
+      environment: 'dev',
+      team: 'test',
+      service: 'foo'
+    }
+
+    const customEnvs = ['dev', 'test']
+    serviceToPagerDutyServiceOverride.foo = { environments: customEnvs }
+
+    const envsToAlert = ['prod', 'management']
+    for (const env of envsToAlert) {
+      alert.environment = env
+      expect(shouldSendAlert(alert, envsToAlert)).toBe(false)
+    }
+
+    for (const env of customEnvs) {
+      alert.environment = env
+      expect(shouldSendAlert(alert, envsToAlert)).toBe(true)
+    }
+  })
```

Comment:
worth adding `expect.hasAssertions()` to make sure you are getting the correct amount of expected assertions you are expecting here - https://jestjs.io/docs/expect#expecthasassertions

Reply:
or `expect.assertions(number)` - https://jestjs.io/docs/expect#expectassertionsnumber